pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -640,8 +640,60 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	if cd.Status.ProvisionRef == nil {
 		if cd.Status.InstallRestarts > 0 && cd.Annotations[tryInstallOnceAnnotation] == "true" {
 			cdLog.Debug("not creating new provision since the deployment is set to try install only once")
+			conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+				cd.Status.Conditions,
+				hivev1.ProvisionStoppedCondition,
+				corev1.ConditionTrue,
+				"InstallOnlyOnceSet",
+				"Deployment is set to try install only once",
+				controllerutils.UpdateConditionIfReasonOrMessageChange)
+			if changed {
+				cd.Status.Conditions = conditions
+				cdLog.Debugf("setting ProvisionStoppedCondition to %v", corev1.ConditionTrue)
+				if err := r.Status().Update(context.TODO(), cd); err != nil {
+					cdLog.WithError(err).Log(controllerutils.LogLevel(err), "failed to update cluster deployment status")
+					return reconcile.Result{}, err
+				}
+			}
+			return reconcile.Result{}, nil
+		}
+		if cd.Spec.InstallAttemptsLimit != nil && cd.Status.InstallRestarts >= int(*cd.Spec.InstallAttemptsLimit) {
+			cdLog.Debug("not creating new provision since the install attempts limit has been reached")
```

Comment:
will including the attempt index and the max retry count in the log help in triage?

Reply:
This information is already available in the ClusterDeployment (cd.spec.InstallAttemptsLimit). I do not think it will add anything useful.